influxql/select.go

```diff
@@ -822,18 +822,40 @@ func buildTransformIterator(lhs Iterator, rhs Iterator, op Token, ic IteratorCre
 			left:  newBufIntegerIterator(left),
 			right: newBufIntegerIterator(right),
 			fn: func(a *IntegerPoint, b *IntegerPoint) *FloatPoint {
-				p := &FloatPoint{
-					Name: a.Name,
-					Tags: a.Tags,
-					Time: a.Time,
-					Aux:  a.Aux,
-				}
-				if (a != nil && b != nil) && (!a.Nil && !b.Nil) {
-					p.Value = fn(a.Value, b.Value)
+				if a != nil && b != nil {
```

Comment:
@thbourlove the idea is to return early and keep indentation to a minimum 😄 . You just need to add an extra condition.

```go
if a == nil && b == nil {
    return nil
} else if a != nil {
    // b is nil
    return &FloatPoint{.....}
} else if b != nil {
    // a is nil
    return &FloatPoint{.....}
}

// neither a or b are nil; bulk of logic here
p := &FloatPoint{
    Name: a.Name,
    Tags: a.Tags,
    Time: a.Time,
    Aux:  a.Aux,
}
.......
```